common/dynamicconfig/nopClient.go

```diff
@@ -31,45 +31,39 @@ import (
 // nopClient is a dummy implements of dynamicconfig Client interface, all operations will always return default values.
 type nopClient struct{}
 
-func (mc *nopClient) GetValue(name Key, defaultValue interface{}) (interface{}, error) {
+func (mc *nopClient) GetValue(name Key) (interface{}, error) {
 	return nil, NotFoundError
 }
 
-func (mc *nopClient) GetValueWithFilters(
-	name Key, filters map[Filter]interface{}, defaultValue interface{},
-) (interface{}, error) {
+func (mc *nopClient) GetValueWithFilters(name Key, filters map[Filter]interface{}) (interface{}, error) {
 	return nil, NotFoundError
 }
 
-func (mc *nopClient) GetIntValue(name Key, filters map[Filter]interface{}, defaultValue int) (int, error) {
-	return defaultValue, NotFoundError
+func (mc *nopClient) GetIntValue(name IntKey, filters map[Filter]interface{}) (int, error) {
+	return name.DefaultInt(), NotFoundError
```

Comment:
here I think `DefaultInt()` is defensible, since `GetValue` returns nil (as it did before).

... though I think "all operations will always return default values" implies it should have been returning `defaultValue` like other clients.  and like some of these methods return, but not others.

... this client was/is kinda messed up when you look closely, huh.

Reply:
it can be deleted later. inMemoryClient can be used to replace nopClient.

Reply:
yeah, that sounds pretty reasonable.  that'd also allow updates, which seems like a good thing for any place where nopClient is used.